test/e2e_node/memory_manager_test.go

```diff
@@ -339,12 +339,17 @@ var _ = SIGDescribe("Memory Manager [Serial] [Feature:MemoryManager]", func() {
 			}, 30*time.Second, framework.Poll).Should(gomega.BeNil())
 
 			ginkgo.By("restarting kubelet to pick up pre-allocated hugepages")
+
 			// stop the kubelet and wait until the server will restart it automatically
 			stopKubelet()
```

Comment:
If you look at the `stopKubelet` func, you see the comment: `stopKubelet will kill the running kubelet, and returns a func that will restart the process again`.

Have you tried using something like:

```go
startKubelet := stopKubelet()
// wait until the kubelet health check will fail
gomega.Eventually(func() bool {
	return kubeletHealthCheck(kubeletHealthCheckURL)
}, time.Minute, time.Second).Should(gomega.BeFalse())

startKubelet()
```

instead?

Reply:
@odinuge yes, i did. it did not work.

Reply:
Do we know why?

Reply:
`systemd` has `active` vs `exited` played some role in it. ( https://askubuntu.com/questions/1304142/systemctl-status-active-exited-vs-running ) i solely depend on the CI for running this, so `restart` basically worked fine.